drivers/scheduler/k8s/k8s.go

```diff
@@ -7517,3 +7528,25 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	clonedSCName := originalStorageClass.Name + "-clone"
+	clonedSCobj, err := k8sStorage.GetStorageClass(clonedSCName)
+	if err == nil {
+		return clonedSCobj.Name, nil
+	} else {
+		log.Infof("Cloned SC with the name %v does not exist and has to be created", clonedSCName)
+		originalStorageClass.Name = clonedSCName
```

Comment:
Mutation of Input Parameter should always be avoided. Please make a copy of the original instead or create new from scratch.

Reply:
Created a copy.